### PR TITLE
Mark tests test_accessor_sliced_datacube and test_grdview_drapegrid_dataarray as xfail

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Closes https://github.com/GenericMappingTools/pygmt/issues/1989.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_accessor.py [hunk 1]
```diff
@@ -5,13 +5,9 @@
```
Comment (weiji14) [inline pygmt/tests/test_accessor.py:5]:
```suggestion
import sys

```

File: pygmt/tests/test_grdview.py [hunk 2]
```diff
@@ -214,6 +214,7 @@ def test_grdview_on_a_plane_styled_with_facadepen(xrgrid):
     return fig
 
 
+@pytest.mark.xfail
```
[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/tests/test_grdview.py:217]: > * Update the baseline image to the one that looks correct on Windows as you mentioned in [Windows test failures for test_accessor_sliced_datacube and test_grdview_drapegrid_dataarray #2062 (comment)](https://github.com/GenericMappingTools/pygmt/issues/2062#issuecomment-1220680290), which you are doing in [WIP: Update the baseline image for test_grdview_drapegrid_dataarray.png #2068](https://github.com/GenericMappingTools/pygmt/pull/2068)
> * Mark the tests as xfail only on LInux and macOS. That way, we still have the Windows unit test available to capture any possible changes.

Actually, the image generated in the Windows CI is also different from the "correct" baseline image as mentioned in #2062, so updating the baseline image still means that the tests fail on all three platforms.

File: pygmt/tests/test_accessor.py [hunk 3]
```diff
@@ -75,9 +76,14 @@ def test_accessor_set_non_boolean():
 
 
 @pytest.mark.skipif(
-    gmt_version < Version("6.4.0"),
+    gmt_version == Version("6.3.0"),
```
Comment (seisman) [inline pygmt/tests/test_accessor.py:79]:
```suggestion
    gmt_version < Version("6.3.0"),
```

File: pygmt/tests/test_accessor.py [hunk 4]
```diff
@@ -75,9 +76,14 @@ def test_accessor_set_non_boolean():
 
 
 @pytest.mark.skipif(
-    gmt_version < Version("6.4.0"),
+    gmt_version < Version("6.3.0"),
```
Comment (seisman) [inline pygmt/tests/test_accessor.py:79]:
```suggestion
    gmt_version < Version("6.4.0"),
```

**Review by weiji14 — APPROVED** [review]

**Pull request merged**